packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -2466,6 +2491,382 @@ class _MenuDirectionalFocusAction extends DirectionalFocusAction {
+        children: <TextSpan>[
+          if (index > 0) TextSpan(text: label.substring(0, index), style: defaultStyle),
+          TextSpan(
+              text: label.substring(index, index + 1),
+              style: defaultStyle.copyWith(decoration: TextDecoration.underline)),
+          if (index < label.length - 1) TextSpan(text: label.substring(index + 1), style: defaultStyle),
+        ],
+      ),
+    );
+  }
+
+  /// Strips out any accelerator markers from the given [label], and unescapes
+  /// any escaped ampersands.
+  ///
+  /// If [setIndex] is supplied, it will be called before this function returns
+  /// with the index in the returned string of the accelerator character.
+  ///
+  /// {@macro flutter.material.menu_anchor.menu_accelerator_label.label}
+  static String stripAcceleratorMarkers(String label, {void Function(int index)? setIndex}) {
+    int quotedAmpersands = 0;
+    final StringBuffer displayLabel = StringBuffer();
+    int acceleratorIndex = -1;
+    for (int i = 0; i < label.length; i += 1) {
+      // Stop looking one before the end, since a single ampersand at the end is
+      // just treated as a quoted ampersand.
+      if (label[i] != '&' || i == label.length - 1) {
+        displayLabel.write(label[i]);
+        continue;
+      }
+      final String acceleratorCharacter = label[i + 1];
+      if (acceleratorIndex == -1 && acceleratorCharacter != '&') {
+        acceleratorIndex = i - quotedAmpersands;
+      }
+      displayLabel.write(acceleratorCharacter);
+      // As we encounter '&&' pairs, the later indices must be adjusted so that
+      // they correspond with indices in the stripped string.
+      quotedAmpersands += 1;
+      i += 1; // Skip over the character after the ampersand.
+    }
+    setIndex?.call(acceleratorIndex);
+    return displayLabel.toString();
+  }
+
+  @override
+  State<MenuAcceleratorLabel> createState() => _MenuAcceleratorLabelState();
+
+  @override
+  String toString({DiagnosticLevel minLevel = DiagnosticLevel.info}) {
+    return '$MenuAcceleratorLabel("$label")';
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(StringProperty('label', label));
+  }
+}
+
+class _MenuAcceleratorLabelState extends State<MenuAcceleratorLabel> {
+  late String _displayLabel;
+  int _acceleratorIndex = -1;
+  MenuAcceleratorCallbackBinding? _binding;
+  _MenuAnchorState? _anchor;
+  ShortcutRegistry? _shortcutRegistry;
+  ShortcutRegistryEntry? _shortcutRegistryEntry;
+  bool _showAccelerators = false;
+
+  @override
+  void initState() {
+    super.initState();
+    if (_platformSupportsAccelerators()) {
+      final Set<LogicalKeyboardKey> keysPressed = RawKeyboard.instance.keysPressed;
+      _showAccelerators = keysPressed.contains(LogicalKeyboardKey.altLeft) ||
+        keysPressed.contains(LogicalKeyboardKey.altRight);
+      RawKeyboard.instance.addListener(_handleKeyEvent);
```

Comment:
Is there no way to do this without RawKeyboard?  I guess you would know if there was a way!